### PR TITLE
17223 detach behavior event

### PR DESCRIPTION
**Pull request by np25071984**
| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | #17223

### CODE REVIEW

File: framework/base/Behavior.php [hunk 1]
```diff
@@ -27,6 +27,10 @@ class Behavior extends BaseObject
      */
     public $owner;
 
+    /**
+     * @var Array Attached events handlers
+     */
+    public $_attachedEvents = [];
```
Comment (laxity7) [inline framework/base/Behavior.php]: Why publicly?

**Comment by samdark** [comment]
Seems this one won't be merged as the issue isn't staightforward at all...

File: framework/base/Behavior.php [hunk 2]
```diff
@@ -85,7 +90,7 @@ public function attach($owner)
     public function detach()
     {
         if ($this->owner) {
-            foreach ($this->events() as $event => $handler) {
+            foreach ($this->_attachedEvents as $event => $handler) {
```
Comment (rob006) [inline framework/base/Behavior.php]: Maybe we should call `unset($this->_attachedEvents[$event]);` after `$this->owner->off()`? Or reset `$_attachedEvents` at the and of `detach()`?

Comment (samdark) [reply framework/base/Behavior.php]: Done.

**Review by rob006 — APPROVED** [review]

**Pull request merged**